validator/validator-main.protoascii

```diff
@@ -3581,33 +3610,23 @@ tags: {
         "password|"
         ")(\\s|$)"
   }
-  attrs: { name: "value" }
-  attrs: { name: "width" }
-  # <amp-bind>
-  attrs: { name: "[accept]" }
-  attrs: { name: "[accesskey]" }
-  attrs: { name: "[autocomplete]" }
-  attrs: { name: "[checked]" }
-  attrs: { name: "[disabled]" }
-  attrs: { name: "[height]" }
-  attrs: { name: "[inputmode]" }
-  attrs: { name: "[max]" }
-  attrs: { name: "[maxlength]" }
-  attrs: { name: "[min]" }
-  attrs: { name: "[minlength]" }
-  attrs: { name: "[multiple]" }
-  attrs: { name: "[pattern]" }
-  attrs: { name: "[placeholder]" }
-  attrs: { name: "[readonly]" }
-  attrs: { name: "[required]" }
-  attrs: { name: "[selectiondirection]" }
-  attrs: { name: "[size]" }
-  attrs: { name: "[spellcheck]" }
-  attrs: { name: "[step]" }
-  attrs: { name: "[type]" }
-  attrs: { name: "[value]" }
-  attrs: { name: "[width]" }
+  attr_lists: "input-common-attr"
+  attr_lists: "input-name-attr"
+  spec_url: "https://www.ampproject.org/docs/reference/components/amp-form"
+}
+tags: {
+  html_format: AMP
+  tag_name: "INPUT"
+  spec_name: "INPUT [type=password]"
+  attrs: {
+    name: "type"
+    value_casei: "password"
+    mandatory: true
+    dispatch_key: NAME_VALUE_DISPATCH
+  }
+  attr_lists: "input-common-attr"
   attr_lists: "input-name-attr"
+  mandatory_ancestor: "FORM [method=POST]"
```

Comment:
We would need runtime validation as well, I don't think `sanitizer.js` version has similar restriction.

Reply:
If I understand correctly, you are suggesting enforcing that "`password` attribute on `input` tag is OK inside of a `form` element"? I think this may be a significant effort in `sanitizer.js`. 

Would it be ok to not render password fields inside of templates for V1?

Reply:
not necessarily in sanitizer (which is going away anyway). How about `amp-form.js` doing `user().assert()` that there is no `password` when it is not post xhr?